File: mili-server/src/main/java/fun/bm/mili/config/modules/experiment/EntityDamageSourceTraceConfig.java
package fun.bm.mili.config.modules.experiment;

import me.earthme.luminol.config.IConfigModule;
import me.earthme.luminol.config.flags.ConfigClassInfo;
import me.earthme.luminol.config.flags.ConfigInfo;
import me.earthme.luminol.enums.EnumConfigCategory;

@ConfigClassInfo(category = EnumConfigCategory.EXPERIMENT, name = "entity_damage_source_trace")
public class EntityDamageSourceTraceConfig implements IConfigModule {
    @ConfigInfo(name = "enabled", comments =
            """
                    允许跨不同区域调度器追踪伤害来源。""")
    public static boolean enabled = false;
}

File: mili-server/src/main/kotlin/fun/bm/mili/config/modules/experiment/EntityDamageSourceTraceConfig.kt
package `fun`.bm.mili.config.modules.experiment

import com.electronwill.nightconfig.core.file.CommentedFileConfig
import `fun`.bm.mili.config.modules.ConfigModule
import me.earthme.luminol.config.flags.ConfigClassInfo
import me.earthme.luminol.config.flags.ConfigInfo
import me.earthme.luminol.enums.EnumConfigCategory

@ConfigClassInfo(category = EnumConfigCategory.EXPERIMENT, name = "entity-damage-source-trace")
class EntityDamageSourceTraceConfig : ConfigModule {
    companion object {
        @ConfigInfo(name = "enabled", comments = "Enable entity damage source trace") @JvmField var enabled = false
    }
    override fun onLoaded(c: CommentedFileConfig) {}
    override fun onUnloaded(c: CommentedFileConfig) {}
}
